python/dgl/nn/pytorch/conv/sageconv.py

```diff
@@ -103,29 +111,38 @@ def forward(self, graph, feat):
             is size of output feature.
         """
         graph = graph.local_var()
-        feat = self.feat_drop(feat)
-        h_self = feat
+
+        if torch.is_tensor(feat):
+            feat_src = feat_dst = self.feat_drop(feat)
+        else:
+            feat_src = self.feat_drop(feat[0])
+            feat_dst = self.feat_drop(feat[1])
+
+        h_self = feat_dst
+
         if self._aggre_type == 'mean':
-            graph.ndata['h'] = feat
+            graph.sdata['h'] = feat_src
             graph.update_all(fn.copy_src('h', 'm'), fn.mean('m', 'neigh'))
-            h_neigh = graph.ndata['neigh']
+            h_neigh = graph.ddata['neigh']
         elif self._aggre_type == 'gcn':
-            graph.ndata['h'] = feat
+            graph.sdata['h'] = feat_src
+            graph.ddata['h'] = feat_dst     # same as above if homogeneous
             graph.update_all(fn.copy_src('h', 'm'), fn.sum('m', 'neigh'))
             # divide in_degrees
             degs = graph.in_degrees().float()
```

Comment:
I am just curious that if we doing sampling, how to define the degree here? the degree in bipartite graph (sampled) or the whole graph?

Reply:
The in degree would be those of destination nodes.